src/components/TodoList.js

```diff
@@ -1,36 +1,53 @@
 import React, { useEffect } from 'react'
 import ListItem from './ListItem'
-import { connect } from 'react-redux'
-import { getTodos } from '../store/todos'
+
+/*
+  TodoList component
+  - Function: TodoList is a presentational component that displays a list of todos as
+              the user adds them
+    Props:    It receives the following props from the Homepage component:
+              - todos - list of todos held in state (Redux store)
+              - editTodo - edits a single todo list item
+              - deleteTodo - deletes a todo from the list
+              - refreshTodos - updates to the current list of todos after a change been made
+              - getTodo - fetches a single todo item
+    Children: ListItem
+                - ListItem will receive as props from TodoList the following:
+                    - all of the properties of each individual todo
+                          - id,
+                          - task,
+                          - completed
+                    - editTodo - edits a single todo item
+                    - deleteTodo - deletes a single todo item
+                    - getTodo - fetches a single todo from the Redux store. This item will also
+                                correspond to the react router route ('/todos/:id) with id being
+                                the id of the todo item
```

Comment:
What's the use case for `ListItem` receiving the `getTodo` prop?

Perhaps it should only receive the `todo` `editTodo` and `deleteTodo`. Those two callbacks should trigger redux to update which should then propagate down to this component via props.

Reply:
I was going to send it down to SingleTodoView to get the todo information from the backend.

>

Reply:
Is `SingleTodoView` meant to be a Page level component or a child component? If Page Level, then it should interact directly with redux using connect. If a child component, then I think it should just receive it's todo data via props. Then it can used anywhere a todo is available

Reply:
ok